geomesa-filter/src/main/scala/org/locationtech/geomesa/filter/FilterHelper.scala

```diff
@@ -10,75 +10,145 @@ package org.locationtech.geomesa.filter
 
 import java.util.Date
 
-import com.vividsolutions.jts.geom.{Geometry, MultiPolygon, Point, Polygon}
-import org.geotools.factory.CommonFactoryFinder
-import org.geotools.geometry.jts.{JTS, ReferencedEnvelope}
-import org.joda.time.{DateTime, DateTimeZone, Interval}
-import org.locationtech.geomesa.filter.visitor.SafeTopologicalFilterVisitorImpl
-import org.locationtech.geomesa.utils.filters.Typeclasses.BinaryFilter
+import com.vividsolutions.jts.geom._
+import org.geotools.filter.spatial.BBOXImpl
+import org.joda.time.{DateTime, DateTimeZone}
 import org.locationtech.geomesa.utils.geohash.GeohashUtils
 import org.locationtech.geomesa.utils.geohash.GeohashUtils._
 import org.locationtech.geomesa.utils.geotools.GeometryUtils
-import org.locationtech.geomesa.utils.text.WKTUtils
+import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
 import org.opengis.feature.simple.SimpleFeatureType
 import org.opengis.filter._
-import org.opengis.filter.expression.{Expression, Literal, PropertyName}
 import org.opengis.filter.spatial._
-import org.opengis.filter.temporal.{After, Before, During, TEquals}
-import org.opengis.temporal.Period
 
 import scala.collection.JavaConversions._
 
 object FilterHelper {
-  // Let's handle special cases with topological filters.
-  def updateTopologicalFilters(filter: Filter, sft: SimpleFeatureType): Filter =
-    filter.accept(new SafeTopologicalFilterVisitorImpl(sft), null).asInstanceOf[Filter]
-
-  def getFirstBinarySpatialOpPropertyName(op: BinarySpatialOperator): PropertyName = op.getExpression1 match {
-    case pn: PropertyName => pn
-    case _                => op.getExpression2 match {
-      case pn: PropertyName => pn
-      case _                =>
-        throw new Exception(s"Neither child of a binary spatial operator was a property name:  $op")
+
+  import org.locationtech.geomesa.utils.geotools.GeometryUtils.{geoFactory => gf}
+  import org.locationtech.geomesa.utils.geotools.WholeWorldPolygon
+
+  val MinDateTime = new DateTime(0, 1, 1, 0, 0, 0, DateTimeZone.UTC).getMillis
+  val MaxDateTime = new DateTime(9999, 12, 31, 23, 59, 59, DateTimeZone.UTC).getMillis
+
+  private val SafeGeomString = "gm-safe"
+
+  /**
+    * Creates a new bbox with valid bounds and attribute
+    *
+    * @param op bbox
+    * @param sft simple feature type
+    * @return valid bbox
+    */
+  def visitBBOX(op: BBOX, sft: SimpleFeatureType): Filter = {
```

Comment:
As a meta-question, is all this thread-safe?

I.e., do we ever process a filter in parallel?  If so, the geometry tweaking might have race conditions...  If not, game on!